src/bin/cli/polkadot/upload.rs

```diff
@@ -0,0 +1,132 @@
+                        code_hash: format!("{:?}", result.code_hash),
+                        deposit: result.deposit,
+                    };
+                    if self.output_json() {
+                        println!("{}", upload_result.to_json()?);
+                    } else {
+                        upload_result.print();
+                        display_dry_run_result_warning("upload");
+                    }
+                }
+                Err(err) => {
+                    let metadata = upload_exec.client().metadata();
+                    let err = ErrorVariant::from_dispatch_error(&err, &metadata)?;
+                    if self.output_json() {
+                        return Err(err);
+                    } else {
+                        name_value_println!("Result", err);
+                    }
+                }
+            }
+        } else {
+            let upload_result = upload_exec.upload_code().await?;
+            let display_events = upload_result.display_events;
+            let output = if self.output_json() {
+                display_events.to_json()?
+            } else {
+                let token_metadata = TokenMetadata::query(upload_exec.client()).await?;
+                display_events
+                    .display_events(self.extrinsic_cli_opts.verbosity()?, &token_metadata)?
+            };
+            println!("{output}");
+            if let Some(code_stored) = upload_result.code_stored {
+                let upload_result = CodeHashResult {
+                    code_hash: format!("{:?}", code_stored.code_hash),
+                };
+                if self.output_json() {
+                    println!("{}", upload_result.to_json()?);
+                } else {
+                    upload_result.print();
+                }
+            } else {
+                let code_hash = hex::encode(code_hash);
+                return Err(anyhow::anyhow!(
+                    "This contract has already been uploaded with code hash: 0x{code_hash}"
+                )
+                .into());
+            }
+        }
+        Ok(())
+    }
+}
+
+#[derive(serde::Serialize)]
+pub struct CodeHashResult {
+    pub code_hash: String,
+}
+
+impl CodeHashResult {
+    pub fn to_json(&self) -> Result<String> {
+        Ok(serde_json::to_string_pretty(self)?)
+    }
+
+    pub fn print(&self) {
+        name_value_println!("Code hash", format!("{:?}", self.code_hash));
+    }
+}
+
+#[derive(serde::Serialize)]
+pub struct UploadDryRunResult {
+    pub result: String,
+    pub code_hash: String,
+    pub deposit: Balance,
+}
+
+impl UploadDryRunResult {
+    pub fn to_json(&self) -> Result<String> {
+        Ok(serde_json::to_string_pretty(self)?)
+    }
+
+    pub fn print(&self) {
+        name_value_println!("Result", self.result);
+        name_value_println!("Code hash", format!("{:?}", self.code_hash));
+        name_value_println!("Deposit", format!("{:?}", self.deposit));
+    }
+}
```

Comment:
This code is stolen from [cargo-contract](https://github.com/paritytech/cargo-contract/blob/632959c5e37e7d86bf09e1966e8c6637321cd9aa/crates/cargo-contract/src/cmd/upload.rs#L123C1-L123C1).